bindings/java/src/org/sleuthkit/datamodel/blackboardutils/ArtifactsHelper.java

```diff
@@ -0,0 +1,180 @@
+	 * @param timeStamp   date/time trackpoint recorded, can be 0 if not
+	 *                    available
+	 * @param name        trackpoint name, can be empty/null if not available
+	 * @param programName name of program that recorded trackpoint , can be
+	 *                    empty or null if not available
+	 *
+	 * @return artifact added
+	 */
+	public BlackboardArtifact addGPSLocation(double latitude, double longitude,
+			long timeStamp, String name, String programName) {
+
+		return addGPSLocation(latitude, longitude, timeStamp, name, programName,
+				Collections.<BlackboardAttribute>emptyList());
+	}
+
+	/**
+	 * Adds a TSK_GPS_TRACKPOINT artifact.
+	 *
+	 * @param latitude            location latitude, required
+	 * @param longitude           location longitude, required
+	 * @param timeStamp           date/time trackpoint recorded, can be 0 if not
+	 *                            available
+	 * @param name                trackpoint name, can be empty/null if not
+	 *                            available
+	 * @param programName         name of program that recorded trackpoint , can
+	 *                            be empty or null if not available
+	 * @param otherAttributesList other attributes, can be empty list of no
+	 *                            additional attributes
+	 *
+	 * @return artifact added
+	 */
+	public BlackboardArtifact addGPSLocation(double latitude, double longitude, long timeStamp, String name, String programName,
+			Collection<BlackboardAttribute> otherAttributesList) {
+
+		BlackboardArtifact gpsTrackpointArtifact = null;
+		try {
+			Collection<BlackboardAttribute> attributes = new ArrayList<>();
+
+			// Create artifact
+			gpsTrackpointArtifact = getAbstractFile().newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACKPOINT);
+
+			// Add basic attributes 
+			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_LATITUDE, getModuleName(), latitude));
+			attributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_LONGITUDE, getModuleName(), longitude));
+
+			addAttributeIfNotZero(timeStamp, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME, attributes);
+
+			addAttributeIfNotNull(name, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME, attributes);
+			addAttributeIfNotNull(programName, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PROG_NAME, attributes);
+
+			// add the attributes 
+			gpsTrackpointArtifact.addAttributes(attributes);
+			gpsTrackpointArtifact.addAttributes(otherAttributesList);
+
+			// post artifact 
+			getSleuthkitCase().getBlackboard().postArtifact(gpsTrackpointArtifact, getModuleName());
+		} catch (TskCoreException ex) {
+			logger.log(Level.SEVERE, "Unable to add GPS trackpoint artifact", ex); //NON-NLS
+			return null;
+		} catch (Blackboard.BlackboardException ex) {
+			logger.log(Level.SEVERE, String.format("Unable to post artifact %s", ((gpsTrackpointArtifact != null) ? gpsTrackpointArtifact.getArtifactID() : "")), ex);  //NON-NLS
+		}
```

Comment:
Low level utility classes should not take responsibility for exception handling policy. This is a project coding guideline. The exceptions should be thrown so that the client code can decide what to do with them. For example, the client module may wish to return an error code from its "process" method.

This comment applies to multiple locations in this PR.